sandy/core/cov.py

```diff
@@ -470,14 +496,27 @@ def invert(self):
         0 0.00000e+00 0.00000e+00 0.00000e+00
         1 0.00000e+00 5.00000e-01 0.00000e+00
         2 0.00000e+00 0.00000e+00 3.33333e-01
+
+        >>> S = sandy.CategoryCov(np.diag(np.array([1, 2, 3])))
+        >>> np.round(S.invert(sparse=True).data,2)
+                    0           1           2
+        0 1.00000e+00 0.00000e+00 0.00000e+00
+        1 0.00000e+00 5.00000e-01 0.00000e+00
+        2 0.00000e+00 0.00000e+00 3.30000e-01
         """
         index, columns = self.data.index, self.data.columns
-        M_nonzero_idxs, M_reduce = self._reduce_size()
-        unit = np.identity(len(M_reduce))
-        M_reduce_inv = splu(csc_matrix(M_reduce)).solve(unit)
-        data = CategoryCov._restore_size(M_nonzero_idxs, M_reduce_inv,
-                                         len(self.data))
-        M_inv = pd.DataFrame(data.data,
+        if sparse is True:
+            cov = self.to_sparse(method='coo_matrix').tocsc()
+            lu = spsl.splu(cov)
+            eye = np.eye(cov.shape[0])
+            data = lu.solve(eye)

```

Comment:
You don't need to reduce and restore?

Reply:
No, because the method is optimized for the inversion. Even if we use spsl.inv(), there is not need to reduce and restore, becuase sparse matrix only contain the non null values.

Reply:
Ok, can ou make sure of this with a test?

Reply:
Yes